### ISSUE TITLE
 "InterfaceMVSNet" Program Aborts with "Core Dumped" Message

### ISSUE THREAD

**Issue by thomas-19** [cdcseacave/openMVS#999]
Description:

Upon running the InterfaceMVSNet, the program is prematurely terminated with a "Core Dumped" message, notably following the return of the main function. Debugging efforts have traced the potential source of the issue to line 296 in "InterfaceMVSNet.cpp" which reads as follows: "Platform& platform = scene.platforms.emplace_back()". Despite attempts to rectify the situation, the problem persists.

Steps to Reproduce:

Download and install OpenMVS.
Execute the build command.
Run InterfaceMVSNet with the following command: "./InterfaceMVSNet -i ~/Desktop/test_lidar_data/scan -o ~/Desktop/test_lidar_data/scan/scene.mvs"
Expected Outcome:

The InterfaceMVSNet program is expected to run smoothly, producing output without the error message: "free(): invalid pointer Aborted (core dumped)".

Actual Outcome:

19:56:52 [App     ] Build date: May 22 2023, 09:00:54
19:56:52 [App     ] CPU: 12th Gen Intel(R) Core(TM) i7-12700 (20 cores)
19:56:52 [App     ] RAM: 31.08GB Physical Memory 2.00GB Virtual Memory
19:56:52 [App     ] OS: Linux 5.13.0-30-generic (x86_64)
19:56:52 [App     ] Disk: 12.02GB (287.37GB) space
19:56:52 [App     ] SSE & AVX compatible CPU & OS detected
19:56:52 [App     ] Command line: InterfaceMVSNet -i /home/thomas/Desktop/test_lidar_data/scan -o /home/thomas/Desktop/test_lidar_data/scan/scene.mvs
19:56:52 [App     ] CUDA device 0 initialized: NVIDIA GeForce RTX 2080 Ti (compute capability 7.5; memory 10.74GB)
19:56:52 [App     ] Scene saved to interface format (0ms):
        19 images (19 calibrated)
        0 points, 0 vertices, 0 faces
19:56:52 [App     ] Imported data: 19 platforms, 19 images, 0 vertices (73ms)
19:56:52 [App     ] MEMORYINFO: {
19:56:52 [App     ]     VmPeak:  5103324 kB
19:56:52 [App     ]     VmSize:  5084712 kB
19:56:52 [App     ] } ENDINFO
free(): invalid pointer
Aborted (core dumped)

**Comment by cdcseacave** [comment]
pls share a dataset for debug

**Comment by thomas-19** [comment]
https://drive.google.com/uc?id=17ZoojQSubtzQhLCWXjxDLznF2vbKz81E&export=download

hi, @cdcseacave . I use the MVSNet authors recommend dtu data.

**Comment by cdcseacave** [comment]
the scan you sent me is invalid: more cameras than images

**Comment by thomas-19** [comment]
@cdcseacave 

Thank you for your response. After eliminating the redundant camera files, and using a new dataset(https://drive.google.com/file/d/1O8ZE19Grai7u9k2tGznBgKT9txBk_2Pr/view?usp=drive_link).
The InterfaceMVSNet is still encountering an issue and crashing. Here are the logs.

Build date: Jun 1, 2023, 14:35:23
CPU: 12th Gen Intel(R) Core(TM) i7-12700 (20 cores)
RAM: 31.08GB Physical Memory, 2.00GB Virtual Memory
OS: Linux 5.13.0-30-generic (x86_64)
Disk: 5.52GB free space (287.37GB total)
CPU & OS: SSE & AVX compatible
Command line: InterfaceMVSNet -i /home/thomas/Desktop/dtu/scan118/ -o /home/thomas/Desktop/dtu/scan118/scene.mvs
CUDA device 0: NVIDIA GeForce RTX 2080 Ti (compute capability 7.5; memory 10.74GB)
Scene saved to interface format (0ms): 49 images (49 calibrated), 0 points, 0 vertices, 0 faces
Imported data: 49 platforms, 49 images, 0 vertices (77ms)
MEMORYINFO:
VmPeak: 5103328 kB
VmSize: 5084716 kB
Error: free(): invalid pointer
Status: Aborted (core dumped)

**Comment by thomas-19** [comment]
@cdcseacave 
Yes, the scene execution is satisfactory. However, the program unexpectedly terminates and it displays the error message: "Error: free(): invalid pointer. Status: Aborted (core dumped)". This occurs precisely at the "return" of the main function.

I try to trace this bug. when I comment out  line 296 in "InterfaceMVSNet.cpp" "Platform& platform = scene.platforms.emplace_back()". the crash log disappear. While I am not able to solve this problem.

**Comment by thomas-19** [comment]
@cdcseacave
Yes, the scene execution is satisfactory.
I am not sure why the problem happens, I will try reinstall OpenMVS in another environment, and try to figure out why.

Thanks

**Comment by cdcseacave** [comment]
InterfaceMVSNet core dump is typically caused by mismatched depth-map/camera resolutions or missing files. Use latest develop branch.
